python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -301,17 +309,26 @@ def get_asset_observations(
     check.opt_int_param(limit, "limit")
     check.opt_float_param(before_timestamp, "before_timestamp")
     check.opt_float_param(after_timestamp, "after_timestamp")
+
     instance = graphene_info.context.instance
-    event_records = instance.get_event_records(
-        EventRecordsFilter(
-            event_type=DagsterEventType.ASSET_OBSERVATION,
-            asset_key=asset_key,
-            asset_partitions=partitions,
-            before_timestamp=before_timestamp,
-            after_timestamp=after_timestamp,
-        ),
-        limit=limit,
-    )
+    cursor = None
+    event_records = []
+    while True:
```

Comment:
same here.... we don't need to do the while loop if we can guarantee a non-None limit (which might just be a graphql change).